src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -579,28 +577,87 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
-        for (DetailNode child : ast.getChildren()) {
-            final String text;
-            if (child.getChildren().length == 0) {
-                text = child.getText();
+        final boolean foundEnd = appendFirstSentence(ast, period, result);
+        if (!foundEnd) {
+            result.setLength(0);
+        }
+        return result.toString();
+    }
+
+    /**
+     * Finds and appends first sentence to result.
+     *
+     * @param ast Javadoc node to append.
+     * @param period Period character.
+     * @param result Builder to append to.
+     * @return true if the period character was found, false otherwise
+     */
+    private static boolean appendFirstSentence(
+            DetailNode ast, String period, StringBuilder result) {
+        boolean foundEnd = false;
+        if (ast.getChildren().length == 0) {
+            final String text = ast.getText();
+            final int periodIndex = findEndingPeriod(text, period);
+            if (periodIndex >= 0) {
+                result.append(text, 0, periodIndex);
+                foundEnd = true;
             }
             else {
-                text = getFirstSentence(child);
+                result.append(text);
             }
+        }
+        for (DetailNode child : ast.getChildren()) {
+            if (appendFirstSentence(child, period, result)) {
+                foundEnd = true;
+                break;
+            }
+        }
+        return foundEnd;
+    }
 
-            if (text.contains(periodSuffix)) {
-                result.append(text, 0, text.indexOf(periodSuffix) + 1);
+    /**
+     * Find position of an ending period in the text. Ignores any period not followed by
+     * whitespace.
+     *
+     * @param text text to search
+     * @param period period character
+     * @return position of period character, or -1 if there is no ending period
+     */
+    private static int findEndingPeriod(String text, String period) {
+        int periodIndex = text.indexOf(period);
+        while (periodIndex >= 0) {
+            final int afterPeriodIndex = periodIndex + period.length();
+            if (isEndOrWhitespace(text, afterPeriodIndex)) {
                 break;
             }
+            else {
+                periodIndex = text.indexOf(period, afterPeriodIndex);
+            }
+        }
+        return periodIndex;
+    }
 
-            result.append(text);
+    /**
+     * Test whether a position in some text is after the end or is whitespace.
+     *
+     * @param text text to test
+     * @param index position to test
+     * @return true if position is at the end or contains a whitespace character
+     */
+    private static boolean isEndOrWhitespace(String text, int index) {
+        final boolean matches;
+        if (index >= text.length()) {
+            matches = true;
         }
-        return result.toString();
+        else {
+            matches = Character.isWhitespace(text.charAt(index));
+        }
+        return matches;
```

Comment:
I don't know how to make the inspections happy here. Right now it's failing an IntelliJ inspection with "If statement can be replaced with '||'", but if I do that it fails Checkstyle with "Avoid inline conditionals."